wire-ios-sync-engine/Source/Repositories/UserRepository_.swift

```diff
@@ -0,0 +1,147 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+import WireDataModel
+
+/// Facilitate access to users related domain objects.
+///
+/// A repository provides an abstraction for the access and storage
+/// of domain models, concealing how and where the models are stored
+/// as well as the possible source(s) of the models.
+
+protocol UserRepositoryProtocol_ {
```

Comment:
I think we can merge this with the existing user repository, which should just mean to add an accessor to the self user.

Reply:
Ok, but that also means we need to start using the UserRepository with all its dependencies in production code, but yeah shouldn't have any side effect if the new methods aren't used.

I'll give it a try later